edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/EdgeletModule.cs

```diff
@@ -83,15 +83,15 @@ public class EdgeletModule : Module
         protected override void Load(ContainerBuilder builder)
         {
             // IModuleClientProvider
-            builder.Register(
-                    c => new ModuleClientProvider(
-                        c.Resolve<ISdkModuleClientProvider>(),
-                        this.upstreamProtocol,
-                        this.proxy,
-                        this.productInfo,
-                        this.closeOnIdleTimeout,
-                        this.idleTimeout,
-                        this.useServerHeartbeat))
+            builder.Register(c => new ModuleClientProvider(
+                    c.Resolve<ISdkModuleClientProvider>(),
+                    c.Resolve<Task<IRuntimeInfoProvider>>(),
```

Comment:
Instead of passing in the Task<IRuntimeInfoProvider>, you can unwrap the Task here to get the IRuntimeInfoProvider and pass that in.. 
See this for an example - https://github.com/Azure/iotedge/pull/6089/files#diff-45b37839c10852faaaf172bd1791f0d56684127ff66998a8e256ebd0ec9a8a26L115

Reply:
That link does not expand to a line reference for me. If the proposal is something like `c.Resolve<Task<IRuntimeInfoProvider>>().GetAwaiter().GetResult()`, I did not use that here since I did not want to run the risk of creating a deadlock.